README.md

```diff
@@ -149,3 +149,103 @@ console and perform any post-boot verification that's required (e.g.
 verify certain packages are installed, etc).
 
 To exit "qemu", one can use "Ctrl-A X".
+
+## Creating new build variants and new roles
+
+This repository contains different build variants which are used to
+generate customized images and leverage different ansible roles to customize
+those variants. The following instructions can be used to create a new
+build variant and roles.
+
+### Step 1. Create and populate the ansible role directory
+
+The new variant that you're planning to create may need some additional
+customized roles. To do so you will need to create the role[s] that the
+variant needs to execute. These roles are located in the
+"live-build/misc/ansible-roles" directory and follow the ansible
+role directory structure (i.e. tasks, handlers, files, etc.)
+
+Create a new directory or directories for the role[s] you will need
+for your variant. In this example, we will create "appliance-build.dcenter"
+to match the name of our variant that we plan to create:
+
+    $ mkdir live-build/misc/ansible-roles/appliance-build.dcenter
+
+With the toplevel directory created, we can populate the directory structure
+with any tasks, files, handlers, templates, etc. In our example, we only
+need "files" and "tasks":
+
+    $ ls -l live-build/misc/ansible-roles/appliance-build.dcenter
+    total 0
+    drwxr-xr-x  3 root  root   96 Jun 19 19:17 files
+    drwxr-xr-x  4 root  root  128 Jun 19 19:17 tasks
+
+There are many roles that exist in "live-build/misc/ansible-roles" and can
+serve as good examples.
+
+### Step 2. Create the new variant directory structure
```

Comment:
These docs are great, but I do wonder if it'd be better to encapsulate all this into a script that we could call to create the new variant directory structure, and base ansible playbook/configuration. Then we could reference that script in these docs.